tools/osx/crosstool/cc_toolchain_config.bzl

```diff
@@ -3842,108 +3842,153 @@ def _impl(ctx):
             ],
         )
     elif (ctx.attr.cpu == "darwin_x86_64"):
-        default_link_flags_feature = feature(
-            name = "default_link_flags",
-            enabled = True,
-            flag_sets = [
-                flag_set(
-                    actions = all_link_actions +
-                              ["objc-executable", "objc++-executable"],
-                    flag_groups = [
-                        flag_group(
-                            flags = ["-no-canonical-prefixes"],
-                        ),
-                    ],
-                ),
+        default_link_flags_feature_flags = [
+            flag_set(
+                actions = all_link_actions +
+                          ["objc-executable", "objc++-executable"],
+                flag_groups = [
+                    flag_group(
+                        flags = ["-no-canonical-prefixes"],
+                    ),
+                ],
+            ),
+            flag_set(
+                actions = [
+                    ACTION_NAMES.cpp_link_dynamic_library,
+                    ACTION_NAMES.cpp_link_nodeps_dynamic_library,
+                ],
+                flag_groups = [flag_group(flags = ["-undefined", "dynamic_lookup"])],
+            ),
+            flag_set(
+                actions = [
+                    ACTION_NAMES.cpp_link_executable,
+                    "objc-executable",
+                    "objc++-executable",
+                ],
+                flag_groups = [flag_group(flags = ["-undefined", "dynamic_lookup"])],
+                with_features = [with_feature_set(features = ["dynamic_linking_mode"])],
+            ),
+        ]
+        if ctx.fragments.cpp.do_not_use_macos_set_install_name:
```

Comment:
instead of these conditions could we only inject the `runtime_solib_name` variable with the feature is enabled?